jakarta-eclipse/org.eclipse.lsp4jakarta.core/src/org/jakarta/jdt/persistence/PersistenceEntityQuickFix.java

```diff
@@ -0,0 +1,152 @@
+ * <ul>
+ * <li> Add a (no-arg) void constructor to this class if the class has other constructors
+ * which do not conform to this
+ * </ul>
+ * 
+ * {@link PersistenceConstants#DIAGNOSTIC_CODE_FINAL_METHODS}
+ * <ul>
+ * <li> Remove the FINAL modifier from all methods in this class
+ * </ul>
+ *
+ * {@link PersistenceConstants#DIAGNOSTIC_CODE_FINAL_VARIABLES}
+ * <ul>
+ * <li> Remove the FINAL modifier from all variables in this class
+ * </ul>
+ * 
+ * {@link PersistenceConstants#DIAGNOSTIC_CODE_FINAL_CLASS}
+ * <ul>
+ * <li> Remove the FINAL modifier from this class
+ * </ul>
+ * 
+ * @author Leslie Dawson (lamminade)
+ *
+ */
+public class PersistenceEntityQuickFix implements IJavaCodeActionParticipant {
+
+    @Override
+    public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+            IProgressMonitor monitor) throws CoreException {
+        ASTNode node = context.getCoveredNode();
+        IBinding parentType = getBinding(node);
+        if (parentType != null) {
+            List<CodeAction> codeActions = new ArrayList<>();
+            
+            // add constructor
+            if (diagnostic.getCode().getLeft().equals(PersistenceConstants.DIAGNOSTIC_CODE_MISSING_EMPTY_CONSTRUCTOR)) {
+                codeActions.addAll(addConstructor(diagnostic, context, parentType));
+            }
+            
+            // remove modifiers
+            if (diagnostic.getCode().getLeft().equals(PersistenceConstants.DIAGNOSTIC_CODE_FINAL_METHODS) 
+                    || diagnostic.getCode().getLeft().equals(PersistenceConstants.DIAGNOSTIC_CODE_FINAL_VARIABLES) 
+                    || diagnostic.getCode().getLeft().equals(PersistenceConstants.DIAGNOSTIC_CODE_FINAL_CLASS)) {
+                codeActions.addAll(removeModifiers(diagnostic, context, parentType));
+            }
+            
+            return codeActions;
+        }
+        return null;
+    }
+    
+    protected static IBinding getBinding(ASTNode node) {
+        if (node.getParent() instanceof VariableDeclarationFragment) {
+            VariableDeclarationFragment fragment = (VariableDeclarationFragment) node.getParent();
+            return ((VariableDeclarationFragment) node.getParent()).resolveBinding();
+        }
+        return Bindings.getBindingOfParentType(node);
+    }
+    
+    private List<CodeAction> addConstructor(Diagnostic diagnostic, JavaCodeActionContext context, IBinding parentType) throws CoreException {
+        List<CodeAction> codeActions = new ArrayList<>();
+
+        // option for protected constructor
+        String name = "Add a no-arg protected constructor to this class";
+        ChangeCorrectionProposal proposal = new AddConstructorProposal(name,
+                context.getCompilationUnit(), context.getASTRoot(), parentType, 0);
+        CodeAction codeAction = context.convertToCodeAction(proposal, diagnostic);
+
+        if (codeAction != null) {
+            codeAction.setTitle(name);
```

Comment:
`setTitle` is handled int he `convertToCodeaAction` method so we can remove this and the following `setTitle()`

Reply:
56c1b47